torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -707,22 +707,30 @@ Tensor prod_backward(
   if (input.dim() == 0) {
     return grad;
   }
+  if (input.numel() == 0) {
+    return input;
+  }
   dim = at::maybe_wrap_dim(dim, input.sizes().size());
   if (!keepdim && input.dim() != 1) {
     grad = grad.unsqueeze(dim);
     result = result.unsqueeze(dim);
   }
   if (input.is_meta()) {
-    return prod_safe_zeros_backward(grad, input, dim);
+    return prod_safe_zeros_backward(grad, input, dim).view_as(input);
   }
 
   Tensor zero_mask = (input == 0);
   Tensor slice_zero_count = zero_mask.sum(dim, true);
-  int64_t total_zeros = slice_zero_count.sum().item<int64_t>();
-  if (total_zeros == 0) {
-    return grad * (result / input).conj();
+  auto total_zeros = slice_zero_count.sum();
+  if (isTensorSubclassLike(total_zeros)) {
+    // For Composite Compliance, always take the safer (and slower) path
+    return prod_safe_zeros_backward(grad, input, dim).view_as(input);
```

Comment:
Can we just do
```
if isTensorSubclassLike(input) {
...
}
```
instead? If so then we can skip the computation of the total_zeros tensor